pkgs/by-name/co/countryguess/package.nix

```
@@ -0,0 +1,48 @@
+{
+  lib,
+  python3Packages,
+  fetchFromGitHub,
+  makeWrapper,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "countryguess";
+  version = "0-unstable-2025-03-04";
+  # upstream pyproject.toml is nonsense. Copied from another project
+  # without customizing it for this project.
+  format = "other";
+
+  src = fetchFromGitHub {
+    owner = "swarbler";
+    repo = "countryguess";
+    rev = "28f45231bc3d8bedeb7d1b51d56ca1b56796ff8c";
+    hash = "sha256-S/fy94aRoVI2CvICrviQ2ZgVESWYLuREb5mwsfXL6Hc=";
+  };
+
+  dependencies = with python3Packages; [
+    art
+    colorama
+  ];
+
+  # upstream python file lacks shebang
+  prePatch = ''
```

```suggestion
  postPatch = ''
```

sed can easily prepnd a line

---

I was having difficulties getting sed to work on both MacOS and Linux as GNUsed is slightly different than BSDsed (which darwin uses).

---

Also, why postPatch vs prePatch? @SuperSandro2000

---

Are we not always using gnused in nix? I am not sure.

prePatch is run before applying any patches and is only useful to eg move files or prepare files. If we do changes in prePatch then patches can break.